src/design_parameters.rs

```diff
@@ -4,3 +4,6 @@ pub const ADC_SETUP_TIME: f32 = 220e-9;
 
 /// The maximum DAC/ADC serial clock line frequency. This is a hardware limit.
 pub const ADC_DAC_SCK_MHZ_MAX: u32 = 50;
+
+/// The optimal counting frequency of the hardware timers used for timestamping and sampling.
+pub const TIMER_FREQUENCY_MHZ: u32 = 100;
```

Comment:
I was avoiding this because the `500.mhz()` is not a `const-fn` so it can't be called in the `const` variables. However, I think we could instead wrap all these up with a manual `MegaHertz{100}` to construct the object instead. I like that idea